Client/lib/gMap.js

```diff
@@ -0,0 +1,282 @@
+
+  var gMap = {}
+  
+  gMap.map;
+  gMap.markers = [];
+  
+  gMap.distance = [];
+  gMap.duration = [];
+  
+  gMap.pathLatLng = [];
+  gMap.travelMode = 'WALKING';
+  gMap.travelButtonList = [];
+
+  gMap.directionsDisplay;
+  gMap.directionsService;
+
+  gMap.startGMap = function (pos){
+    //can be -> BICYCLING, DRIVING, TRANSIT, WALKING
+    //----------------------------------
+    //creatig the Map
+    //----------------------------------
+    var mapOptions = {
+      center: pos,
+      zoom: 12,
+      backgroundColor: '#568',
+      disableDoubleClickZoom : true,
+      disableDefaultUI: true,
+      keyboardShortcuts: false
+    };
+    gMap.map = new google.maps.Map(document.getElementById('gMap'),mapOptions);
+    
+    //----------------------------------
+    //styling the Map
+    //----------------------------------
+
+    var mapStyles = [
+      {
+        stylers: [ { hue: "#00ffe6" }, { saturation: -20 } ]
+      },{
+        featureType: "road",
+        elementType: "geometry",
+        stylers: [ { lightness: 100 }, { visibility: "simplified" } ]
+      },{
+        featureType: "road",
+        elementType: "labels",
+        stylers: [ { visibility: "off" } ]
+      }
+    ];
+
+    var styledMap = new google.maps.StyledMapType(mapStyles,{name: "Styled Map"});
+    gMap.map.mapTypes.set('map_style', styledMap);
+    gMap.map.setMapTypeId('map_style');
+
+    //----------------------------------
+    //events
+    //----------------------------------
+    google.maps.event.addListener(gMap.map, 'click', function(event) {
+      //makeMarker(event.latLng);
+      if(gMap.pathLatLng.length < 10){
+        gMap.pathLatLng.push(event.latLng);
+      }
+      gMap.createPath();
+    });
+    //----------------------------------
+    //adding custom GUI elements
+    //----------------------------------
+
+    new gMap.CenterControl('TOP_LEFT', '-', function(){
+      gMap.map.setZoom(gMap.map.getZoom()-1);
+    });
+
+    new gMap.CenterControl('TOP_LEFT', '+', function(){
+      gMap.map.setZoom(gMap.map.getZoom()+1);
+    });
+
+    new gMap.CenterControl('BOTTOM_LEFT', 'W', function(){
+      gMap.travelMode = 'WALKING';
+    },true,true);
+
+    new gMap.CenterControl('BOTTOM_LEFT', 'B', function(){
+      gMap.travelMode= 'BICYCLING';
+    },true);
+
+    new gMap.CenterControl('BOTTOM_LEFT', 'D', function(){
+      gMap.travelMode= 'DRIVING';
+    },true);
+
+    new gMap.CenterControl('BOTTOM_LEFT', 'T', function(){
+      gMap.travelMode= 'TRANSIT';
+    },true);
+
+  }
+
+  //------------------------
+  //functions
+  //------------------------
+
+  gMap.CenterControl = function (position,text, callback, addToList,startAsActive) {
+
+    // Set CSS for the control border
```

Comment:
Not unless we use something like browserify. You can ignore my comment regarding this I am just tunnel visioned on the back end.